### PR TITLE
created max-circular-subArray-sum directory

### LINKED ISSUE #1728: Maximum circular subarray sum in C and C++.
I would like to work on this.
Assign me this issue. :)

### PR DESCRIPTION
**Pull request by MdAkdas**
Fixes #1728

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- New algorithm

#### Languages Used:
- C
- C++

#### Files Added:
- max_circula_subarray_sum.c
- max_circula_subarray_sum.cpp

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by MdAkdas** [comment]
My first PR in this project. :)
please review the PR @jainaman224 .

File: Max_Circular_SubArray_Sum/max_circula_subarray_sum.c [hunk 1]
```diff
@@ -0,0 +1,71 @@
+//Que-Given n numbers (both +ve and -ve), arranged in a circle,
```
Comment (jainaman224) [inline Max_Circular_SubArray_Sum/max_circula_subarray_sum.c]: Typo in file name.

File: Max_Circular_SubArray_Sum/max_circula_subarray_sum.c [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+// return max of two no.
+int max(int num1, int num2)
+{
+    return (num1 > num2 ) ? num1 : num2;
+}
+
+// Function to find contiguous sub-array with the largest sum
+// in given set of integers
+int kadaneAlgo(int a[],int n)
```
Comment (jainaman224) [inline Max_Circular_SubArray_Sum/max_circula_subarray_sum.c]: Indentation is not right throughout the file.

**Comment by jainaman224** [comment]
Please read about code convention. Like spaces around the operators etc.

File: Max_Circular_SubArray_Sum/max_circular_subArray_sum.c [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+/*Function to find contiguous sub-array with the largest sum
+in given set of integers*/
+int kadaneAlgo(int a[], int n)
+{  
+    int max_so_far = 0, max_ending_here = 0;
+
+    for(int i = 0; i < n ; i++)
+ 	{ 
+		max_ending_here += a[i];
```
Comment (jainaman224) [inline Max_Circular_SubArray_Sum/max_circular_subArray_sum.c]: Incorrect indentation. Fix indentation throughout the file.

File: Max_Circular_SubArray_Sum/max_circular_subArray_sum.c [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+}
+
+/*Function to find contiguous sub-array with the largest sum
+in given set of integers*/
+int kadaneAlgo(int a[], int n)
+{  
+    int max_so_far = 0, max_ending_here = 0;
+
+    for(int i = 0; i < n ; i++)
+ 	{
```
Comment (jainaman224) [inline Max_Circular_SubArray_Sum/max_circular_subArray_sum.c]: It should align with start of above line.

**Comment by MdAkdas** [comment]
I've messed up in squashing the commits.
that's why i have made fresh PR request.
please review.
https://github.com/jainaman224/Algo_Ds_Notes/pull/2354/files

**Pull request closed without merging**